README.md
budgie-indicator-applet
-----------------------

AppIndicator applet for budgie-desktop.

*Help required to resolve the TODO and ENHANCEMENTS lists*


To compile:

    sudo apt install git libtool dpkg-dev intltool libtool libgtk-3-dev libido3-0.1-dev libindicator3-dev libpeas-dev budgie-core-dev
    
    
    git clone https://github.com/ubuntubudgie/budgie-indicator-applet
    cd budgie-indicator-applet
    ./autogen.sh --prefix=/usr --libdir=/usr/lib
    make
    sudo make install

To run: install the recommended packages

    sudo apt install gir1.2-appindicator3-0.1
    
    budgie-panel --replace &
    
    Use Raven to add the applet to the panel.
    

TODO
-----

 - [x] Applet background needs to use panel colour for inbuilt-theme 
 - [x] Code cleanup - copyright statements, unused code, change boilerplate budgie-applet
 - [x] Correct debian/copyright
 - [x] check all the build dependencies are actually needed

Enhancements
-----

 - [ ] For the applet settings add capability to change indicator order
 - [x] Appindicator spacing is too wide when not using built-in-theme
 - [x] Applet background needs to respect raven stylise regions option
 - [x] Applet background needs to use panel colour for user-defined theme
 - [ ] When used with the system-tray applet hide the network applet icon rather than permanently hiding
 - [ ] Change from using GtkMenu and GtkMenuItem to GtkButtonBox/GtkButton and therefore allow GtkPopover when button click
 
 *Tips for Development*
 
 Use the following to run GTK Inspector - use to investigate CSS and other properties
 
     GTK_DEBUG=interactive budgie-panel --replace
     
 Use the following to print out g_debug messages
 
     G_MESSAGES_DEBUG=Indicator-Applet-DEBUG budgie-panel --replace


CHANGELOG.md
# Changelog

## [Unreleased](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/HEAD)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/0.7.1...HEAD)

**Closed issues:**

- Merge Commits from XFCE and Mate indicators [\#29](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/29)
- Bug\#998730: budgie-indicator-applet: 0.7.1-1 [\#28](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/28)

## [0.7.1](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/0.7.1) (2021-10-28)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.7.0...0.7.1)

**Closed issues:**

- the old oversized icon issue is back [\#27](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/27)

## [v0.7.0](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.7.0) (2021-06-12)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.6.2...v0.7.0)

**Closed issues:**

- The top bar increases height when the Slack icon appears on AppIndicador Applet. [\#25](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/25)
- \(K\)SNI support and google chrome hangouts extension [\#22](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/22)
- does not know how to scale SVG \(or QT ?\) icon? - \(was: Albert icon - too large\) [\#18](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/18)

## [v0.6.2](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.6.2) (2019-12-04)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.6.1...v0.6.2)

**Fixed bugs:**

- Open appindicator Using Right-Click, doesn't perform any action! [\#12](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/12)

**Closed issues:**

- Unable to kill System Apps [\#21](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/21)

## [v0.6.1](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.6.1) (2019-06-04)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.6...v0.6.1)

**Implemented enhancements:**

- Port to libayatana-indicator [\#11](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/11)

**Closed issues:**

- App Icons don't scale down if panel is under 37px [\#19](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/19)
- Debian Buster - build against revised ayatana library [\#17](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/17)
- Wireless signal strength icon not visible on default background color [\#16](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/16)

**Merged pull requests:**

- Panelsize [\#20](https://github.com/UbuntuBudgie/budgie-indicator-applet/pull/20) ([fossfreedom](https://github.com/fossfreedom))

## [v0.6](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.6) (2018-04-14)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.5...v0.6)

**Merged pull requests:**

- Support Ayatana Indicators and Ubuntu Indicators [\#13](https://github.com/UbuntuBudgie/budgie-indicator-applet/pull/13) ([sunweaver](https://github.com/sunweaver))

## [v0.5](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.5) (2018-02-16)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.4...v0.5)

**Closed issues:**

- Vertically oriented panel [\#10](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/10)
- background color different than panel [\#9](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/9)

## [v0.4](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.4) (2017-08-20)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.3...v0.4)

**Fixed bugs:**

- Cannot remove appindicator via raven [\#2](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/2)
- dropbox causes panel and system crash [\#1](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/1)

**Closed issues:**

- How to get Budgie show certain indicator icons? [\#8](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/8)
- dropbox does not display [\#6](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/6)
- System apps not showing; Dropbox is just black square icon [\#4](https://github.com/UbuntuBudgie/budgie-indicator-applet/issues/4)

## [v0.3](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.3) (2016-12-02)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.2...v0.3)

## [v0.2](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.2) (2016-11-11)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/v0.1...v0.2)

## [v0.1](https://github.com/ubuntubudgie/budgie-indicator-applet/tree/v0.1) (2016-11-06)

[Full Changelog](https://github.com/ubuntubudgie/budgie-indicator-applet/compare/b56d201da85d9821bd234a3eab728e2734d9af26...v0.1)



\* *This Changelog was automatically generated by [github_changelog_generator](https://github.com/github-changelog-generator/github-changelog-generator)*


src/eggaccelerators.h
/* eggaccelerators.h
 * Copyright (C) 2002  Red Hat, Inc.
 * Developed by Havoc Pennington
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Library General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Library General Public License for more details.
 *
 * You should have received a copy of the GNU Library General Public
 * License along with this library; if not, write to the
 * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
 */

#ifndef __EGG_ACCELERATORS_H__
#define __EGG_ACCELERATORS_H__

#include <gtk/gtk.h>

G_BEGIN_DECLS

/* Where a value is also in GdkModifierType we coincide,
 * otherwise we don't overlap.
 */
typedef enum {
        EGG_VIRTUAL_SHIFT_MASK = 1 << 0,
        EGG_VIRTUAL_LOCK_MASK = 1 << 1,
        EGG_VIRTUAL_CONTROL_MASK = 1 << 2,

        EGG_VIRTUAL_ALT_MASK = 1 << 3, /* fixed as Mod1 */

        EGG_VIRTUAL_MOD2_MASK = 1 << 4,
        EGG_VIRTUAL_MOD3_MASK = 1 << 5,
        EGG_VIRTUAL_MOD4_MASK = 1 << 6,
        EGG_VIRTUAL_MOD5_MASK = 1 << 7,

#if 0
  GDK_BUTTON1_MASK  = 1 << 8,
  GDK_BUTTON2_MASK  = 1 << 9,
  GDK_BUTTON3_MASK  = 1 << 10,
  GDK_BUTTON4_MASK  = 1 << 11,
  GDK_BUTTON5_MASK  = 1 << 12,
  /* 13, 14 are used by Xkb for the keyboard group */
#endif

        EGG_VIRTUAL_META_MASK = 1 << 24,
        EGG_VIRTUAL_SUPER_MASK = 1 << 25,
        EGG_VIRTUAL_HYPER_MASK = 1 << 26,
        EGG_VIRTUAL_MODE_SWITCH_MASK = 1 << 27,
        EGG_VIRTUAL_NUM_LOCK_MASK = 1 << 28,
        EGG_VIRTUAL_SCROLL_LOCK_MASK = 1 << 29,

        /* Also in GdkModifierType */
        EGG_VIRTUAL_RELEASE_MASK = 1 << 30,

        /*     28-31 24-27 20-23 16-19 12-15 8-11 4-7 0-3
         *       7     f     0     0     0    0    f   f
         */
        EGG_VIRTUAL_MODIFIER_MASK = 0x7f0000ff

} EggVirtualModifierType;

gboolean egg_accelerator_parse_virtual(const gchar *accelerator, guint *accelerator_key,
                                       EggVirtualModifierType *accelerator_mods);
void egg_keymap_resolve_virtual_modifiers(GdkKeymap *keymap, EggVirtualModifierType virtual_mods,
                                          GdkModifierType *concrete_mods);
void egg_keymap_virtualize_modifiers(GdkKeymap *keymap, GdkModifierType concrete_mods,
                                     EggVirtualModifierType *virtual_mods);

G_END_DECLS

#endif /* __EGG_ACCELERATORS_H__ */
